### PR TITLE
vendor: Expo SDK 16

### PR DESCRIPTION
**Pull request by inyono**
Updates to React Native 0.43 and Expo SDK 16. Also creates helpers for playing sounds since they change the API again 😬. @Knorrke review shouldn't be needed for most parts (except maybe `helpers/audio.js`) but please check if this works on Windows (after removing `node_modules` and updating Expo on your phone / simulator)

### CODE REVIEW

File: src/components/exercises/ShowLetter.js [hunk 1]
```diff
@@ -26,26 +26,7 @@ const styles = {
...
-  sounds: [letterSound, repeatSound],
-  isRepeat
-}) => {
-  const play = () => {
-    if (isRepeat) this.icon.unfocus();
-    letterSound.playAsync();
-    letterSound.setPlaybackFinishedCallback(() => {
-      letterSound.setPositionAsync(0);
-      if (isRepeat) {
-        this.icon.focus();
```
Comment (Knorrke) [inline src/components/exercises/ShowLetter.js]: This effect is missing in the new version, we probably need an callback for this in the playAll helper function.

File: src/components/helpers/audio.js [hunk 2]
```diff
@@ -2,6 +2,29 @@ import { Audio } from 'expo';
...
+
+  sound.setCallback(status => {
+    if (status.didJustFinish) {
+      sound.setPositionAsync(0);
+      cb && cb();
+    }
+  });
+};
+
+export const playAll = (sounds, delay = 1000) => {
```
Comment (Knorrke) [inline src/components/helpers/audio.js]: as mentioned above: probably we need a callback prop here too

**Comment by inyono** [comment]
Yeah, let's postpone that merge until facebook/react-native#12885 is resolved.

**Comment by inyono** [comment]
Thanks for reviewing :)

**Comment by inyono** [comment]
superseded by #116

**Pull request closed without merging**